src/wallet/wallet.cpp

```diff
@@ -3425,13 +3457,24 @@ void CWallet::ConnectScriptPubKeyManNotifiers()
 
 void CWallet::LoadDescriptorScriptPubKeyMan(uint256 id, WalletDescriptor& desc)
 {
+    bool ranged = desc.descriptor->IsRange();
     if (IsWalletFlagSet(WALLET_FLAG_EXTERNAL_SIGNER)) {
         auto spk_manager = std::unique_ptr<ScriptPubKeyMan>(new ExternalSignerScriptPubKeyMan(*this, desc));
         m_spk_managers[id] = std::move(spk_manager);
     } else {
         auto spk_manager = std::unique_ptr<ScriptPubKeyMan>(new DescriptorScriptPubKeyMan(*this, desc));
         m_spk_managers[id] = std::move(spk_manager);
     }
+
+    // Cache scriptPubKeys if the script is not ranged
+    ScriptPubKeyMan* spkm = m_spk_managers.at(id).get();
+    m_uncached_spkms.insert(spkm);
+    if (!ranged) {
+        for (const auto& script : spkm->GetScriptPubKeys()) {
+            m_cached_spks[script].insert(spkm);
+            m_uncached_spkms.erase(spkm);
+        }
+    }
```

Comment:
```suggestion
    if (!ranged) {
        for (const auto& script : spkm->GetScriptPubKeys()) {
            m_cached_spks[script].insert(spkm);
        }
    }
    else {
        m_uncached_spkms.insert(spkm);
    }
```

Reply:
I prefer the way it was written as it ensures that no spkm will be accidentally missed.